no-std code coverage collection doesn't work

grcov --branch --guess-directory-when-missing --ignore-not-existing --llvm --binary-path target/release/final-elf.bin --source-dir ./ --output-type html --output-path target/release/coverage test.profraw
warning: /home/liuwei/td-shim/test.profraw: unrecognized instrumentation profile encoding format
error: no profile can be merged
01:30:04 [ERROR] Error while executing llvm tools: Failure while running "/home/liuwei/.rustup/toolchains/nightly-2022-05-15-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata" "merge" "-f" "-" "-sparse" "-o" "/tmp/.tmpUGgS7t/1/grcov.profdata"

--- a/td-payload/src/bin/example/main.rs
+++ b/td-payload/src/bin/example/main.rs
@@ -77,6 +77,7 @@ pub extern "C" fn main() -> ! {
             buffer.as_ptr() as u64,
             coverage_len
         );
+        loop {}
     }
 
     panic!("td-payload: all tests finished and enters dead loop");

--- a/td-payload/src/mm/dma.rs
+++ b/td-payload/src/mm/dma.rs
@@ -60,7 +60,7 @@ pub unsafe fn alloc_dma_pages(num: usize) -> Option<usize> {
         .map(|ptr| ptr.as_ptr() as usize)
         .ok()?;
 
-    core::slice::from_raw_parts_mut(addr as *mut u8, SIZE_4K).fill(0);
+    core::slice::from_raw_parts_mut(addr as *mut u8, size).fill(0);
 
     Some(addr)
 }

